### ISSUE TITLE
An error of example/pinn_forward/Helmholtz_Dirichlet_2d_HPO.py

### ISSUE THREAD

**Issue by lijialin03** [lululxvi/deepxde#1446]
### Discussed in https://github.com/lululxvi/deepxde/discussions/1445

<div type='discussions-op-text'>

<sup>Originally posted by **lijialin03** August 18, 2023</sup>
Function 'gp_minimize' of package 'skopt (scikit-optimize)' is used in this example and an error has been reported like,
![16e8051a98747fcc62553d09c11b873b](https://github.com/lululxvi/deepxde/assets/124568209/a4c21d4a-4170-4fd1-a7f0-20c038522b20)
The reason is that 'np.int' used in skopt was deprecated in NumPy 1.20. The newest NumPy is version 1.25.
My skopt version is 0.9.0. I think it is the latest version in https://pypi.org/search/?q=scikit-optimize. So this example may require NumPy 1.20 or lower while there is no version requirements of NumPy in requirements.txt.</div>

**Comment by lululxvi** [comment]
How about fixing this using the solutions at https://github.com/scikit-optimize/scikit-optimize/issues/1138

**Comment by lijialin03** [comment]
> How about fixing this using the solutions at [scikit-optimize/scikit-optimize#1138](https://github.com/scikit-optimize/scikit-optimize/issues/1138)

Yes, the solution do works. I submit a PR here https://github.com/lululxvi/deepxde/pull/1450 to apply it.
